### ISSUE TITLE
Different POIs in different map styles

### ISSUE THREAD

**Issue by rindlerblabla** [osmandapp/Osmand#3736]
I want to view different POIs (the ones you select in the map menu) when viewing the map in bicycle style compared to when driving. E.g. compressed air is interesting when going by bicycle, not when going by car.

Right now it's possible to choose whether to view public transport on car/bicycle/pedestrian map or not but not POIs. My suggestion is to add this feature.

**Comment by xmd5a2** [comment]
What do you mean under "map menu"?
2667217a7f57aafe8169ec3b619243349c9e9fbe

**Comment by rindlerblabla** [comment]
Everything is in Swedish but I guess it's called configure map. The one you reach by clicking on the globe in the upper left corner...

**Comment by xmd5a2** [comment]
You can create your own POI filter. Search - Category - Custom search - select POI - Show - Button "filters" - three dots - Save filter. It will appear in category list.

**Comment by rindlerblabla** [comment]
Yes, I get it. But then it will appear in all "map styles". For example I want the custom made POI filter for compressed air for bicyclea to ONLY be seen when the map is configured for bicycles, not when the map is configured for pedestrian or cars.

**Comment by rindlerblabla** [comment]
Same opinion here https://groups.google.com/forum/m/#!topic/osmand/qaVNGNaTzE8

**Comment by mase76** [comment]
Same here. When configure a POI filter in one view, it is the same in all views.
But this seens to be a bug. All other items are configurable per view, except the POIs.

**Comment by GRBurst** [comment]
That would be a very nice feature and I think it results in the same solution as #2619 ?!

**Comment by nekohayo** [comment]
Oh please yes. For years I've been confused and annoyed at how un-impactful switching map profiles is when POIs (and the other display settings under the _"Show"_ section) are not affected by the map style, and numerous other users reported this in one variant or another (for example in issue #4048, #2704, #2619, #2556, #1276, etc.). And before you say "but only a few people requested this", I'd say most users (of any software in the world) don't care enough to go through the effort to find the bug tracker, have a github account, know how to find the right ticket (the only non-closed one) among the dozen similarly-sounding ones, and make their case here; it takes a special kind of stubborness to go through that :)

The various icons at the top of the "Configure map" switcher screen _look_ like tabs, and they _look_ like they would be global "modes" that affect all the settings that follow them, and so from a usability perspective the users would tend to expect that they behave like that: mode switches. And the whole point of those modes should be to be a quick and _easy_ way to switch between different contexts of using osmand; don't ask me to tack on some other obscure feature like custom filtering for what should be built-into a mode switching UI that led you into thinking it was meant for this.

Everytime I use OSMAnd, it's for a different transportation style when travelling. It could be any of the four main types (car, bike, walk, public transit); in each case I care about completely different things and manually setting the POIs every single time is a big pain in the butt:

* In a car: I care about Filing/charging stations, parking, and "Personal transport" (car rentals). That's it.
* On a bike: I care about "Healthcare" (in case of accident), and enabling the "Contour Lines" and "Hillshade layer"; but those make no sense for me in a car.
* By walking: now I may care about tourism, sightseeing, or food related things, depending on the purpose of my walk.
* By public transit: I'm interested in "public transport" POIs and the "Transport" overlays (ex: transport stops)

It's super inefficient to have to set all the POIs, Transport overlay settings, hillshade/contour lines, everytime I switch between "Car", "Bike", "Walk" and "Public transit" modes. Without those being saved "per context-type", for me there is almost no point in the existence of the separate profiles/modes UI because you end up having to reset everything all the time anyway...

**Comment by pebogufi** [comment]
Again - it is easy and a very powerful feature to create custom search filter.
You can use different of them (include multi select) to use in different profiles and for other use. One of the really great features of osmand.

**Comment by snouup** [comment]
> Same here. When configure a POI filter in one view, it is the same in all views.
But this seens to be a bug. All other items are configurable per view, except the POIs.

I see the opposite in current version. POI overlays are saved by profile. This issue can be closed then, can't it?
